### DISCUSSION TITLE
Thinking about filtering audio at input

### DISCUSSION THREAD

**Discussion by DaveBKC** [mcguirepr89/BirdNET-Pi#451] [General]
I live in a suburb close enough to traffic that I pick up a fairly steady rumble in my audio. I changed the cap value on the AOM-5024P to clean that up a bit, but I'm considering adding a 24db lowpass filter to remove some of the unneeded hiss. It doesn't look like there's anything of value above 8k.

Has anyone tried something like this? Do we know if the software _already_ does a bandpass when analyzing the samples? If it already does, then I won't bother with it.  

Dave
https://app.birdweather.com/stations/278

**Comment by CaiusX** [comment]
Hi Dave

The software processes the wav file unmodified. It would be interesting to see what difference filtering makes. My worry is that it will filter out birds vocalizing in traffic noise frequency ranges. 

Best
AP

**Comment by nilspupils** [comment]
I like the idea of experimenting with filters (mainly for sound quality of the recordings, but the effects on detection are also interesting).
My plan is like this:
1. Set up a "standard" sound file with a big variety of birdsounds
2. Collect noise samples (city, wind, traffic etc.) 
3. Add different amounts of noise to birdsounds
4. Compare detection pre- and post noise (someone has done this already, who was it?) 
5. Apply different filters (in software) and compare the detection pre- and post filter. 

I think i will start with the birdsong standard file next week. Any ideas which european species to include? 
Best, Nils

**Reply by ehpersonal38** [reply]
> Compare detection pre- and post noise (someone has done this already, who was it?)

Here! https://github.com/mcguirepr89/BirdNET-Pi/discussions/358#discussioncomment-2939257

It was kind of a pain because analyze.py expects a certain file format. i.e. `2022-07-18-birdnet-09:49:46.wav`

**Reply by nilspupils** [reply]
Great you have already tested this Ethan! What kind of noise spectra did you use and would you provide them to the (upcoming) noise database?

**Reply by ehpersonal38** [reply]
I just used Audacity's built-in noise generator:
![audacity_kzTWlju3VJ](https://user-images.githubusercontent.com/103586016/179567884-7e6f4194-3f8b-4f86-94c0-305faeb0152b.png)

Can't remember the exact values I used unfortunately. Looking forward to seeing your results though.
